File: src/main/java/io/nightfis/createstellarexoduscore/filter/IsFriendlyFilter.java
package io.nightfis.createstellarexoduscore.filter;

import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.LivingEntity;

public class IsFriendlyFilter extends TargetFilter {

    private static final String ID = "IsFriendlyFilter";

    @Override
    public String getId() {
        return ID;
    }

    @Override
    public boolean match(LivingEntity entity, String arg) {
        return entity.getType().getCategory().isFriendly();
    }

    @Override
    public Component title() {
        return Component.translatable("screen.createstellarexoduscore.auto_aim_turret.filter.is_friendly");
    }
}

File: src/main/kotlin/io/nightfis/createstellarexoduscore/filter/IsFriendlyFilter.kt
package io.nightfis.createstellarexoduscore.filter

import net.minecraft.network.chat.Component
import net.minecraft.world.entity.LivingEntity

class IsFriendlyFilter : TargetFilter() {

    override fun getId(): String = ID

    override fun match(entity: LivingEntity, arg: String): Boolean {
        return entity.type.category.isFriendly
    }

    override fun title(): Component {
        return Component.translatable("screen.create_stellar_exodus_core.auto_aim_turret.filter.is_friendly")
    }

    companion object {
        private const val ID = "IsFriendlyFilter"
    }
}
